usage/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -0,0 +1,465 @@
+
+        if (params != null) {
+            mergeConfigs(configs, params);
+        }
+
+        String aggregationRange = configs.get("usage.stats.job.aggregation.range");
+        String timeZoneStr = configs.get("usage.aggregation.timezone");
+
+        if (timeZoneStr == null) {
+            timeZoneStr = "GMT";
+        }
+        _usageTimezone = TimeZone.getTimeZone(timeZoneStr);
+
+        _aggregationDuration = Integer.parseInt(aggregationRange);
+        if (_aggregationDuration < UsageUtils.USAGE_AGGREGATION_RANGE_MIN) {
+            s_logger.warn("Usage stats job aggregation range is to small, using the minimum value of " + UsageUtils.USAGE_AGGREGATION_RANGE_MIN);
+            _aggregationDuration = UsageUtils.USAGE_AGGREGATION_RANGE_MIN;
+        }
+        s_logger.info("Usage timezone = " + _usageTimezone + " AggregationDuration=" + _aggregationDuration);
+
+        return true;
+    }
+
+    @Override
+    public boolean start() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Starting Quota Manager");
+        }
+        _pid = Integer.parseInt(System.getProperty("pid"));
+        return true;
+    }
+
+    @Override
+    public boolean stop() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Stopping Quota Manager");
+        }
+        return true;
+    }
+
+    @Override
+    public boolean calculateQuotaUsage() {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        boolean jobResult = false;
+        TransactionLegacy txn = TransactionLegacy.open(TransactionLegacy.USAGE_DB);
+        try {
+            // get all the active accounts for which there is usage
+            List<AccountVO> accounts = _accountDao.listAll();
+            for (AccountVO account : accounts) { // START ACCOUNT
+                Pair<List<? extends UsageVO>, Integer> usageRecords = null;
+                List<QuotaUsageVO> quotalistforaccount = new ArrayList<QuotaUsageVO>();
+                do {
+                    s_logger.info("Account =" + account.getAccountName());
+                    usageRecords = _usageDao.getUsageRecordsPendingQuotaAggregation(account.getAccountId(), account.getDomainId());
+                    s_logger.debug("Usage records found " + usageRecords.second());
+                    for (UsageVO usageRecord : usageRecords.first()) {
+                        BigDecimal aggregationRatio = new BigDecimal(_aggregationDuration).divide(s_minutesInMonth, 8, RoundingMode.HALF_EVEN);
+                        switch (usageRecord.getUsageType()) {
+                        case QuotaTypes.RUNNING_VM:
+                            quotalistforaccount.addAll(updateQuotaRunningVMUsage(usageRecord, aggregationRatio));
+                            break;
+                        case QuotaTypes.ALLOCATED_VM:
+                            quotalistforaccount.add(updateQuotaAllocatedVMUsage(usageRecord, aggregationRatio));
```

Comment:
``QuotaTypes.RUNNING_VM`` and ``QuotaTypes.ALLOCATED_VM`` appear to be the same.  Why not consolidate them?